node/impl/full/sync.go

```diff
@@ -105,6 +106,26 @@ func (a *SyncAPI) SyncIncomingBlocks(ctx context.Context) (<-chan *types.BlockHe
 	return a.Syncer.IncomingBlocks(ctx)
 }
 
+func (a *SyncAPI) SlashFilterMinedBlock(ctx context.Context, bh *types.BlockHeader) (*types.BlockHeader, error) {
+	parent, err := a.Syncer.ChainStore().GetBlock(ctx, bh.Parents[0])
+	if err != nil {
+		return nil, nil
+	}
+	err = a.SlashFilter.MinedBlock(ctx, bh, parent.Height)
+	if err != nil && strings.Contains(err.Error(), "mining faults") {
```

Comment:
I think we should refactor `MinedBlock` to return the `others`, instead of parsing the error message here.